app/models/portal/student.rb

```diff
@@ -16,6 +16,10 @@ class Portal::Student < ActiveRecord::Base
   has_many :student_clazzes, :dependent => :destroy, :class_name => "Portal::StudentClazz", :foreign_key => "student_id"
 
   has_many :clazzes, :through => :student_clazzes, :class_name => "Portal::Clazz", :source => :clazz
+  has_many :teachers, :through => :clazzes, :class_name => "Portal::Teacher", :source => :teachers, :uniq => true
+  # students cohorts are infered from its teacher(s)
+  has_many :cohorts, :through => :teachers, :class_name => "Admin::Cohort", :source => :cohorts, :uniq => true
+  has_many :projects, :through => :cohorts, :class_name => "Admin::Project", :source => :project, :uniq => true
```

Comment:
I wasn't sure about plural and singular forms (e.g. `cohorts` => `cohorts` vs `projects` => `project`). It seems to work. :wink: